app/workers/regenerate_image_worker.rb

```diff
@@ -0,0 +1,26 @@
+class RegenerateImageWorker
+  include Sidekiq::Worker
+
+  def perform doc_ids
+    documents = Document.find(doc_ids)
+    documents.each { |doc| doc.update! processed: false } unless documents.blank?
+    processing = false
+    current = doc_ids.pop
+    while !doc_ids.blank?
+      if processing
+        if Document.find(current).processed
+          Rails.logger.info "Regenerated images for document with id #{current},
```

Comment:
Creating a second async worker (RegenerateImageWorker) was an attempt to prevent a single bad document from crashing the whole regeneration process of > 1000 documents. Do you see a way to wrap this logic into PdfToImagesWorker that can add redundancy?